handler.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+#ifndef SABER_HANDLER_HPP
+#define SABER_HANDLER_HPP
+
+//std
+#include <optional>
+
+namespace saber {
+
+template<typename T>
```

Comment:
You need a document-comment here saying:
"What `ValueHandler` does, and why/when you would use it"